File: desafio-03/flafmg/lua/palindromos.lua
function eh_palindromo(num)
    local str_num = tostring(num)
    local str_reversa = string.reverse(str_num)
    return str_num == str_reversa
end

function imprimir_palindromos(inicio, fim)
    for i = inicio, fim do
        if eh_palindromo(i) then
            io.write(i .. " ")
        end
    end
    print()
end

print("digite o numero inicial:")
local inicio = tonumber(io.read())
print("digite o numero final:")
local fim = tonumber(io.read())

imprimir_palindromos(inicio, fim)

File: desafio-03/flafmg/csharp/palindromos.cs
    using System;

    class Program
    {
        static void Main()
        {
            Console.WriteLine("Digite o número inicial:");
            int start = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine("Digite o número final:");
            int end = Convert.ToInt32(Console.ReadLine());

            PrintPalindromes(start, end);
        }

        static bool IsPalindrome(int number)
        {
            string strNumber = number.ToString();
            char[] charArray = strNumber.ToCharArray();
            Array.Reverse(charArray);
            string reversedStr = new string(charArray);
            return strNumber == reversedStr;
        }

        static void PrintPalindromes(int start, int end)
        {
            for (int i = start; i <= end; i++)
            {
                if (IsPalindrome(i))
                {
                    Console.Write(i + " ");
                }
            }
            Console.WriteLine();
        }
    }
